securedrop-admin

```diff
@@ -0,0 +1,159 @@
+SITE_CONFIG = os.path.join(ANSIBLE_PATH, "group_vars/all/site-specific")
+VENV_DIR = os.path.join(SD_DIR, ".venv")
+VENV_ACTIVATION = os.path.join(VENV_DIR, 'bin/activate_this.py')
+
+
+def logger_setup(verbose=False):
+    """ Configure logging handler """
+    # Set default level on parent
+    sdlog.setLevel(logging.DEBUG)
+    level = logging.DEBUG if verbose else logging.INFO
+
+    stdout = logging.StreamHandler(sys.stdout)
+    stdout.setFormatter(logging.Formatter('%(levelname)s: %(message)s'))
+    stdout.setLevel(level)
+    sdlog.addHandler(stdout)
+
+def activate_venv():
+    """Use to activate the local virtualenv"""
+    try:
+        execfile(VENV_ACTIVATION, dict(__file__=VENV_ACTIVATION))
+    except IOError:
+        sdlog.error("Pre-requisites not in place, re-run command with 'setup'")
+        sys.exit(1)
+
+
+def sdconfig(args):
+    """Configure SD site settings"""
+    activate_venv()
+    subprocess.check_call([os.path.join(ANSIBLE_PATH,'configure')],
+                            cwd=ANSIBLE_PATH, env=os.environ.copy())
+
+
+def envsetup(args):
+    """Install SD pre-reqs"""
+    # virtualenv doesnt exist? Install dependencies and create
+    if not os.path.exists(VENV_ACTIVATION):
+        sdlog.info("Installing dependencies. You'll be prompted for the "+\
+            "temporary Tails root credentials. This was set on boot-up screen")
+        sdlog.debug(subprocess.check_output(['sudo', 'su', '-c',
+                                            "apt-get update && \
+                                            apt-get install -y \
+                                            python-virtualenv \
+                                            python-pip \
+                                            ccontrol \
+                                            virtualenv \
+                                            libffi-dev \
+                                            libssl-dev \
+                                            libpython2.7-dev"]))
+
+        # Technically you can create a virtualenv from within python
+        # but pip can only be run over tor on tails, and debugging that
+        # along with instaling a third-party dependency is not worth
+        # the effort here.
+        sdlog.info("Setting up virtualenv")
+        sdlog.debug(subprocess.check_output(['torify', 'virtualenv',
+                                        VENV_DIR], stderr=subprocess.STDOUT))
+
+    pip_install_cmd = [
+            'torify',
+            os.path.join(VENV_DIR, 'bin', 'pip'),
+            'install',
+            ]
+    requirements_cmd = ['-r', os.path.join(ANSIBLE_PATH, 'requirements.txt'),
+                        '--require-hashes']
+    upgrade = ['-U', '--upgrade-strategy', 'only-if-needed']
+
+    sdlog.info("Checking pip dependencies")
+    pip_output = subprocess.check_output(pip_install_cmd+upgrade+requirements_cmd)
+
+    if "Successfully installed" in pip_output:
+        sdlog.info(pip_output)
```

Comment:
If the `-v` switch is mostly a developer debugging convenience (as it seems), then it makes sense to log "Python dependencies successfully upgraded" to info here, and to log `pip_output` to debug regardless of its contents.